### PR TITLE
Refactor CI

### PR DESCRIPTION
**Pull request by masa213f**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -301,21 +317,15 @@ workflows:
       - deploy_github:
           requires:
             - build_release
-          filters:
```
Comment (ymmt2005) [inline .circleci/config.yml]: Why did you remove this?

File: .circleci/config.yml [hunk 2]
```diff
@@ -157,6 +163,7 @@ jobs:
           command: |
             git config --global user.email "neco@cybozu.com"
             git config --global user.name "cybozu-neco"
+      # Check difference between [`release` branch] and [`master` branch + `artifacts_release.go`]
       - run: git checkout release
       - run: git merge --no-commit origin/${CIRCLE_BRANCH}
```
Comment (ymmt2005) [inline .circleci/config.yml]: Fix this as well.

File: .circleci/config.yml (hunk 1 above)
Comment (masa213f) [reply .circleci/config.yml]: Because `filters` is set in the dependent `build_release` job.
Should I specify it explicitly?

File: .circleci/config.yml (hunk 2 above)
Comment (masa213f) [reply .circleci/config.yml]: Thank you! d1ce99a

File: .circleci/config.yml (hunk 1 above)
Comment (ymmt2005) [reply .circleci/config.yml]: I suppose so.
